PHP-Math-Functions-max()-#2063 (#2083)
Codecademy/docs: content/php/concepts/math-functions/terms/max/max.md

---

> Title: 'Max()'

Reviewer: ```suggestion
Title: 'max()'
```

> Description: 'Used to find the highest value in an array or a list of variables. It returns the highest value from the input provided.'

Reviewer: ```suggestion
Description: 'Finds and returns the highest value in an array or a list of variables.'
```

Subjects:
  - 'Computer Science'
  - 'Web Design'
  - 'Web Development'
Tags:
 - 'Functions'
CatalogContent:
  - 'learn-php'
  - 'paths/computer-science'
---

The **`max()`** function in PHP is used to find the highest value in an [array](https://www.codecademy.com/resources/docs/php/arrays) or a list of [variables](https://www.codecademy.com/resources/docs/php/variables). It returns the highest value from the input provided.

## Syntax

```pseudo
max($var1, $var2, $var3, ...);
```

> `$var1`, `$var2`, `$var3` are the [variables](https://www.codecademy.com/resources/docs/php/variables) or [array](https://www.codecademy.com/resources/docs/php/arrays) from which the maximum value will be returned.

Reviewer: ```suggestion
`$var1`, `$var2`, `$var3` are the [variables](https://www.codecademy.com/resources/docs/php/variables) or [arrays](https://www.codecademy.com/resources/docs/php/arrays) from which the maximum value will be returned.
```

> It can be called either with multiple parameters or with a single array parameter.

Reviewer: ```suggestion
It can be called either with multiple parameters or with a single [array](https://www.codecademy.com/resources/docs/php/array) parameter.
```

When called with multiple parameters, the **`max()`** function takes a comma-separated list of variables or values, like this:

```php
$max_value = max($var1, $var2, $var3);

> ```

Reviewer: This is already mentioned in the previous syntax block. Maybe you would like to replace this with an array

Reviewer: Or remove this section because it looks like the following portion speaks on having a single array as the parameter

In this case, the **`max()`** function compares the values of $var1, $var2, and $var3, and returns the highest value among them, which is then assigned to the variable $max_value.

When called with a single array parameter, the **`max()`** function takes an array of values, like this:

```php
$max_value = max([$var1, $var2, $var3]);

```

> In this case, the **`max()`** function compares the values in the array `[$var1, $var2, $var3]`, and returns the highest value among them, which is then assigned to the variable `$max_value`.

Reviewer: ```suggestion
In this case, the **`max()`** function compares the values in the array `[$var1, $var2, $var3]`, returns the highest value among them, and then assigns it to the variable `$max_value`.
```

> It's important to note that if the **`max()`** function is called with a single non-array parameter, like this:

Reviewer: ```suggestion
It's important to note that if the **`max()`** function is called with a single non-array parameter, it will cause an error.:
```

```php
$max_value = max($var1);
```

> it will cause a PHP error, because the function expects at least two parameters when called with multiple values. Therefore, to use **`max()`** with a single value, it's necessary to put it into an array:

Reviewer: ```suggestion
The function expects at least two parameters when called with multiple values. Therefore, to use **`max()`** with a single parameter, it's necessary to use an array:
```

```php
$max_value = max([$var1]);
```

This will return the value of `$var1`, since it's the only value in the array.

## Example

The example uses the `max()` function to find the highest value in the array of numbers and then prints the result using the `echo` statement. The output will be the maximum value found in the array.

```php
$numbers = array(23, 12, 35, 67, 8, 90);
$max_value = max($numbers);
echo "The maximum value in the array is: " . $max_value;
```

> This example results in the following output.

Reviewer: ```suggestion
This example results in the following output:
```
